packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java

```diff
@@ -122,6 +139,112 @@ public ExoPlayer getExoPlayer() {
+    List<ExoPlayerAudioTrackData> audioTracks = new ArrayList<>();
+
+    // Get the current tracks from ExoPlayer
+    Tracks tracks = exoPlayer.getCurrentTracks();
+
+    // Iterate through all track groups
+    for (int groupIndex = 0; groupIndex < tracks.getGroups().size(); groupIndex++) {
+      Tracks.Group group = tracks.getGroups().get(groupIndex);
+
+      // Only process audio tracks
+      if (group.getType() == C.TRACK_TYPE_AUDIO) {
+        for (int trackIndex = 0; trackIndex < group.length; trackIndex++) {
+          Format format = group.getTrackFormat(trackIndex);
+          boolean isSelected = group.isTrackSelected(trackIndex);
+
+          // Create audio track data with metadata
+          ExoPlayerAudioTrackData audioTrack =
+              new ExoPlayerAudioTrackData(
+                  (long) groupIndex,
+                  (long) trackIndex,
+                  format.label,
+                  format.language,
+                  isSelected,
+                  format.bitrate != Format.NO_VALUE ? (long) format.bitrate : null,
+                  format.sampleRate != Format.NO_VALUE ? (long) format.sampleRate : null,
+                  format.channelCount != Format.NO_VALUE ? (long) format.channelCount : null,
+                  format.codecs != null ? format.codecs : null);
+
+          audioTracks.add(audioTrack);
+        }
+      }
+    }
+    return new NativeAudioTrackData(audioTracks);
+  }
+
+  @UnstableApi
+  @Override
+  public void selectAudioTrack(long groupIndex, long trackIndex) {
+    if (trackSelector == null) {
+      Log.w("VideoPlayer", "Cannot select audio track: track selector is null");
+      return;
+    }
+
+    try {
+
+      // Get current tracks
+      Tracks tracks = exoPlayer.getCurrentTracks();
+
+      if (groupIndex >= tracks.getGroups().size()) {
+        Log.w(
+            "VideoPlayer",
+            "Cannot select audio track: groupIndex "
+                + groupIndex
+                + " is out of bounds (available groups: "
+                + tracks.getGroups().size()
+                + ")");
+        return;
+      }
+
+      Tracks.Group group = tracks.getGroups().get((int) groupIndex);
+
+      // Verify it's an audio track and the track index is valid
+      if (group.getType() != C.TRACK_TYPE_AUDIO || (int) trackIndex >= group.length) {
+        if (group.getType() != C.TRACK_TYPE_AUDIO) {
+          Log.w(
+              "VideoPlayer",
+              "Cannot select audio track: group at index "
+                  + groupIndex
+                  + " is not an audio track (type: "
+                  + group.getType()
+                  + ")");
+        } else {
+          Log.w(
+              "VideoPlayer",
+              "Cannot select audio track: trackIndex "
+                  + trackIndex
+                  + " is out of bounds (available tracks in group: "
+                  + group.length
+                  + ")");
+        }
+        return;
+      }
```

Comment:
I don't see the benefit of doing this nested sort of check. Instead, I suggest we warn first if it's not even an audio track and second if the track index isn't valid:

```suggestion
        if (group.getType() != C.TRACK_TYPE_AUDIO) {
          Log.w(
              "VideoPlayer",
              "Cannot select audio track: group at index "
                  + groupIndex
                  + " is not an audio track (type: "
                  + group.getType()
                  + ")");
           return;
        }
        if ((int) trackIndex >= group.length) {
          Log.w(
              "VideoPlayer",
              "Cannot select audio track: trackIndex "
                  + trackIndex
                  + " is out of bounds (available tracks in group: "
                  + group.length
                  + ")");
          return;
      }
```